lib/IRGen/IRGenSIL.cpp

```diff
@@ -4752,8 +4752,7 @@ void IRGenSILFunction::visitDebugValueInst(DebugValueInst *i) {
   if (VarDecl *Decl = i->getDecl()) {
     DbgTy = DebugTypeInfo::getLocalVariable(
         Decl, RealTy, getTypeInfo(SILVal->getType()));
-  } else if (i->getFunction()->isBare() && !SILTy.hasArchetype() &&
-             !VarInfo->Name.empty()) {
+  } else if (!SILTy.hasArchetype() && !VarInfo->Name.empty()) {
```

Comment:
why the change with isBare?

Reply:
A function is marked as bare if it's parsed from SIL. But I also saw cases where the function is coming from Swift code but the instruction (i.e. `DebugValueInst *I`) doesn't have an associated VarDecl. For example, the debug_value instruction was generated during the salvageDebugInfo. In those cases, we will restore the `DbgTy` from `RealTy` and `SILTy`.

Reply:
Under what circumstances does salvageDebugInfo drop the VarDecl? Shouldn't it carry it forward from the salvaged debug_value?

Reply:
We're actually retrieving a VarDecl of a debug_value from its attached SILLocation (`getLoc().getAsASTNode<VarDecl>();`). And in the case of debug_value created by salvageDebugInfo, its SILLocation is "<compiler generated>", which I believe we can not get it as VarDecl.

Reply:
Thanks! We discussed this offline and the piece of info I was missing that we are storing the type info in the new auxiliary info record, and this change effectively allows us to access that.